### PR TITLE
Monotype Random Battle: Enable Monotype-specific movesets

### PR DESCRIPTION
**Pull request by scpinion**
I goobered up the last PR with merge conflicts. After a bit of googling and trying what others suggested in the dev room, I couldn't figure out how to fix it. I just made a new PR.

Contains: 
randomMonotypeTeam in scripts.js
ability to add custom movesets from format-data.js to randomSet in scripts.js
a couple sets for Monotype Random Battles in formats-data.js to serve as examples

### CODE REVIEW

File: data/scripts.js [hunk 1]
```diff
@@ -1174,6 +1174,23 @@ exports.BattleScripts = {
 		}
 
 		let movePool = (template.randomBattleMoves ? template.randomBattleMoves.slice() : Object.keys(template.learnset));
+
+		// Check for alternate move pool.
```
Comment (Zarel) [inline data/scripts.js]: This is perhaps the wrong position for it. This also runs a useless `Object.keys(template.learnset)` before replacing `movePool`, and would be best off being rewritten not to need that.

File: data/scripts.js [hunk 2]
```diff
@@ -1174,6 +1174,23 @@ exports.BattleScripts = {
...
+		if (template[alternateMovePool]) {
+			let newMovePool = template[alternateMovePool].slice();
+
+			// Confirm it is a valid list of moves.
```
Comment (Zarel) [inline data/scripts.js]: ...??? Why?

Do this in the tests if you have to; revalidating every time a battle starts is entirely the wrong decision...

**Comment by TheImmortal** [comment]
You don't need randomMonotypeTeam if we're going with moves in FormatsData. It's just an unnecessary copy ofrandomTeam.

**Comment by scpinion** [comment]
Yes, it is mostly a copy. There are two reasons why I'd like to keep it, though. 

1. There are differences in the NFEs and checks for forms. For example, the Pokemon pool for each type only has one Arceus form and it shouldn't have the 1/18th chance like in standard randbats.

2. In the future, I'd like to make the chance of pulling an uber to be (roughly) equal between the types. Currently, types like Dragon or Flying has a much higher chance than other types and that kinda sucks when you're up against them.

Would you rather me add that stuff in randomTeam?

**Comment by Zarel** [comment]
Are my code review comments unclear?

**Comment by TheImmortal** [comment]
Fine then. It's a good amount of changes from randomTeam.

**Pull request closed without merging**